features/268-next-cf/api.md

```diff
@@ -666,3 +666,139 @@ function. Controller will check the mapping and create fission function with
 k8s secrets. Fission support reading secrets value from files. We will update
 the fission environment runtime to convert those values to environment variable,
 so skygear user can access those secret form env.
+
+## Batch deploy items in skygear.yaml
+
+`skycli app deploy` will deploy all deployment items in `skygear.yaml`.
+`skycli app deploy [ITEM_NAME]` will only create or update item with name
+`[ITEM_NAME]`.
+
+When user use `skycli app deploy`, they need to have all items in the
+`skygear.yaml`. The deployment will not keep the previously deployed item, if
+they don't exist in `skygear.yaml`. e.g. User had deployed function 1 to 4. If
+user run deploy with a config file that only has function 1 to 2. The new
+version app will only have function 1 to 2. User has responsibility to keep the
+full version of `skygear.yaml` and codes.
+
+User can create or update particular item by `skycli app deploy [ITEM_NAME]`.
+e.g. User had deployed function 1 to 4. If user use
+`skycli app deploy function5` to deploy function 5, the new app version will
+have function 1 to 5 after deployment.
+
+### Implementation Details
+
+#### Models
+
+**`deployment`**
+
+Every deployment (one or multiple items) will create a new deployment.
+
+- `id`
+- `created_at`: deployment time
+- `app_id`: skygear app id
+- `version`: deployment version, uuid or hash
+- `tag`: deployment tag, the live version will have tag with `latest`
+- `status`: deployment status (pending, running, deploy failed, stopping, stopped, stop failed)
+
+**`deployment_cloud_code`**
+
+deployment to cloud_code relationship
+
+- `deployment_id`
+- `cloud_code_id`
+
+**`cloud_code_route`**
+
+Denormalized routing table for gateway routing.
+
+- `id`
+- `created_at`
+- `version`: deployment version, will be the same as `version` of `deployment`
+- `path`: user defined path
+- `target_path`: skygear internal path for cloud code routing 
+- `app_id`: skygear app id
+- `backend_url`: cloud code backend url
+- `tag`: deployment tag, the live version will have tag with `latest`
+
+**`static_asset` (TBD)**
+
+
+#### Deployment and Cloud Code Status
+
+Deployment and cloud code status values: pending, running, deploy failed, stopping, stopped, stop failed
```

Comment:
just curious, so our clodecode backend (in this case Fission) will have a webhook / pubsub whatever to call our controller to update the status?

Reply:
We are now polling the status by using fission client (calling the fission controller server) during the deployment task.